hack/deployer/runner/gcloud.go

```diff
@@ -0,0 +1,84 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package runner
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	GCPDir                 = ".gcp"
+	ServiceAccountFilename = "osServiceAccount.json"
+)
+
+// authToGCP authenticates the deployer to the Google Cloud Platform as a service account or as a user.
+func authToGCP(
+	vaultInfo *VaultInfo, vaultPath string, serviceAccountVaultFieldName string,
+	asServiceAccount bool, useNonDefaultCloudSDKPath bool, configureDocker bool, gCloudProject interface{},
+) error {
+	if asServiceAccount {
+		if vaultInfo == nil {
+			return errors.New("vault info not present in the plan to authenticate to GCP")
+		}
+
+		log.Println("Authenticating as service account...")
+
+		client, err := NewClient(*vaultInfo)
+		if err != nil {
+			return err
+		}
+
+		gcpDir := filepath.Join(os.Getenv("HOME"), GCPDir)
+		keyFileName := filepath.Join(gcpDir, ServiceAccountFilename)
+		_ = os.MkdirAll(gcpDir, os.ModePerm)
```

Comment:
nit: any reason to not check the error here ?

Reply:
I didn't reevalute this code as it got moved but no I don't see a good reason.